src/components/app/check-font-loaded.js

```diff
@@ -0,0 +1,69 @@
+import { UPDATE_FONT_LOADED } from '../../actions';
+
+/**
+ * Prevent chart from displaying until the webfont has loaded,
+ * to ensure that text label SVGRect BBox measurements are accurate
+ * @param {Object} store Redux store
+ * @return {Promise} Resolves when font is detected as having loaded, or after 8 seconds
+ */
+const checkFontLoaded = store =>
+  new Promise(resolve => {
+    const { fonts } = document;
+    const fontName = '10px Titillium Web';
+    /**
+     * Dispatch an event to show the chart
+     */
+    const setLoaded = () => {
+      store.dispatch({ type: UPDATE_FONT_LOADED, fontLoaded: true });
+      resolve();
+    };
+
+    // If FontFaceSet is not supported then default to true immediately
+    if (!Boolean(fonts && fonts.check)) {
+      setLoaded();
+      return;
+    }
+
+    /**
+     * Check whether the webfont is ready, and show the chart if so
+     */
+    const checkIfLoaded = () => {
+      if (fonts.check(fontName)) {
+        // Add 0.1s delay because fonts.check often returns true too early
+        setTimeout(setLoaded, 100);
```

Comment:
Is this delay always consistent? e.g. if it loaded after 101 ms

Reply:
It was really hard to say how consistent it is, as it was a trial and error thing - the only way of programmatically detecting it is broken, which is the reason for the delay. It seemed to come in around 50ms or less, so 100ms seemed like a safe delay - long enough to cover the issue, but short enough that it's a pretty imperceptible increase in load-time.